rust/origen/src/core/utility/logger.rs

```diff
@@ -153,14 +138,15 @@ impl Logger {
     }
 
     pub fn default_output_dir() -> PathBuf {
+        let pb;
         if STATUS.is_app_present {
-            return (STATUS.root).to_path_buf().join("log");
+            pb = (STATUS.root).to_path_buf().join("log");
         } else {
-            match env::var("HOME") {
-                Ok(var) => PathBuf::from(var).join(".origen").join("log"),
-                Err(_e) => panic!("No environment variable for HOME!"),
-            }
+            pb = (STATUS.home).to_path_buf().join(".origen").join("log");
         }
+        // create all missing directories to avoid panics
+        fs::create_dir_all(pb.as_path()).expect("Could not create the log directory");
```

Comment:
Should we include the directory that it is trying to make in the error message?
Any panics like this from Rust will just show the user this message and nothing else, so they need to be as descriptive and helpful as possible.